server/Controllers/PointsController.cs

```diff
@@ -52,63 +52,56 @@ public PointsController(IMemoryCache memoryCache, ILogger<PointsController> logg
 
         // GET /api/points
         [HttpGet]
-        public IActionResult GetAll(DateTime startDate)
+        public IActionResult GetAll(string startDate)
         {
             var points = _memoryCache.Get<Dictionary<Guid, Point>>(_pointsKey);
-            return Json(points.Values.Where(point => startDate == null || point.Date >= startDate));
+            var tags = _memoryCache.Get<Dictionary<string, Tag>>(_tagsKey);
+            return Json(
+                points.Values
+                    .Where(point => startDate == null || point.Date >= new HutelDate(startDate))
+                    .Select(p => p.ToJson(tags))
+                    .ToList());
         }
 
-        // POST /api/points
-        [HttpPost]
+        // PUT /api/points
+        [HttpPut]
         [ValidateModelState]
-        public IActionResult PostAll([FromBody]List<Point> replacementPoints)
+        public IActionResult PutAll([FromBody]PointsStorageJson replacementPoints)
         {
             var points = _memoryCache.Get<Dictionary<Guid, Point>>(_pointsKey);
             var tags = _memoryCache.Get<Dictionary<string, Tag>>(_tagsKey);
+            IEnumerable<Point> pointsList;
             try
             {
-                foreach (var point in replacementPoints)
-                {
-                    if (!tags.ContainsKey(point.TagId))
-                    {
-                        throw new ValidationException($"Unknown tag: {point.TagId}");
-                    }
-                    PointValidator.Validate(point, tags[point.TagId]);
-                }
+                pointsList = replacementPoints.Select(p => Point.FromJson(p, tags));
             }
-            catch(ValidationException ex)
+            catch(PointValidationException ex)
             {
                 return new BadRequestObjectResult(ex.ToString());
             }
             points.Clear();
-            foreach (var p in replacementPoints)
+            foreach (var p in pointsList)
             {
                 points.Add(p.Id, p);
```

Comment:
Same ids cand be added to Dictionary
[
{
    "id": "044af706-92e7-435c-b788-cd8370938a11",
   ...
  },
  {
    "id": "044af706-92e7-435c-b788-cd8370938a11",
...
  }
]